tfjs-core/src/io/weights_loader.ts

```diff
@@ -245,3 +235,179 @@ export function weightsLoaderFactory(
     return weightsTensorMap;
   };
 }
+
+type BufferRange = {
+  start: number,
+  end: number,
+  buffer: ArrayBuffer,
+};
+
+export class CompositeArrayBuffer {
```

Comment:
This will be used in another PR that enables large model weights to be stored in a list of ArrayBuffers. That's why it's exported here.